api/graphql/schema.py

```diff
@@ -181,6 +196,10 @@ def resolve_equipment_category_by_pk(self, info, **kwargs):
         pk = kwargs.get("pk")
         return get_object_or_404(EquipmentCategory, pk=pk)
 
+    def resolve_purpose_by_pk(self, info, **kwargs):
+        pk = kwargs.get("pk")
+        return get_object_or_404(Purpose, pk=pk)
+
```

Comment:
I think we don't need this anymore since we got rid of `purpose_by_pk` in the schema.